### PR TITLE
Stabilize the `Option::replace` method

### PR DESCRIPTION
**Pull request by Kerollmops**
This PR stabilize the `Option::replace` method propose in https://github.com/rust-lang/rust/issues/51998.

### CODE REVIEW

File: src/libcore/option.rs [hunk 1]
```diff
@@ -880,7 +878,7 @@ impl<T> Option<T> {
     /// assert_eq!(old, None);
     /// ```
     #[inline]
-    #[unstable(feature = "option_replace", issue = "51998")]
+    #[stable(feature = "option_replace", since = "1.30.0")]
```
Comment (Centril) [inline src/libcore/option.rs]: 1.30.0 seems unlikely? See https://forge.rust-lang.org/index.html for dates.

**Comment by Centril** [comment]
r? @Kimundi 

(Already fcp merged + completed in tracking issue.)

File: src/libcore/option.rs (hunk 1 above)
Comment (Kerollmops) [reply src/libcore/option.rs]: It is unlikely to be merged before `Fri Oct 26 2018` ? You prefer `1.31.0` ?

Comment (Centril) [reply src/libcore/option.rs]: It has to go into beta first ;) i.e. 1.31.0 goes into beta on 2018-10-26.

Comment (Kerollmops) [reply src/libcore/option.rs]: Ok I will change to `1.31.0` so but how do you know the `1.31.0` beta is set to this date ?

Comment (Kerollmops) [reply src/libcore/option.rs]: Ho it starts when the previous version enter stable ? ok ! :)

**Pull request merged**